streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -685,13 +685,13 @@ public boolean test(final K1 key, final V1 value) {
                                                                                     windows.maintainMs());
 
             KStreamKStreamJoin<K1, R, V1, V2> joinThis = new KStreamKStreamJoin<>(otherWindow.name(),
-                                                                                  windows.before,
                                                                                   windows.after,
+                                                                                  windows.before,
```

Comment:
Which doc are you referring to that has the SQL representation? For me I'm just reading the java doc string of `before`, which states as

```
Specifies that records of the same key are joinable if their timestamps are within
the join window interval, and if the timestamp of a record from the secondary stream is
earlier than or equal to the timestamp of a record from the first stream.
```

According to this I think it should be the opposite case, where a record from `stream1`, or the primary stream at time `t` is joinable with other records with sam key with timestamp in `t-before, t`; and a record from `stream2`, or the secondary stream at time `t` is joinable with other records with sam key with timestamp in `t, t + before`. Or if you want to conceptually translate the DSL

```
stream1.join(stream2, joiner, windows.before(before).after(after))
```

it should be translated to

```
SELECT * FROM stream1, stream2
WHERE
  stream1.key = stream2.key
  AND
  stream1.ts - before >= stream2.ts
  AND
  stream1.ts + after <= stream2.ts
```

Re: whether window should be closed or open. There are already some discussions before about making all window specs (join window, time window for aggregations) consistent, will try to dig it out and refer here.